camel/utils/token_counting.py

```diff
@@ -255,6 +257,36 @@ def count_tokens_from_messages(self, messages: List[OpenAIMessage]) -> int:
         return num_tokens
 
 
+
+class AnthropicTokenCounter(BaseTokenCounter):
+
+    def __init__(self, model_type: ModelType):
+        r"""Constructor for the token counter for Anthropic models.
+
+        Args:
+            model_type (ModelType): Model type for which tokens will be
+                counted.
+        """
+
+        self.model_type = model_type
+        self.client = Anthropic()
+        self.tokenizer = self.client.get_tokenizer()
+
+    def count_tokens_from_messages(self, messages: List[OpenAIMessage]) -> int:
+        r"""Count number of tokens in the provided message list using
+        loaded tokenizer specific for this type of model.
+
+        Args:
+            messages (List[OpenAIMessage]): Message list with the chat history
+                in OpenAI API format.
+
+        Returns:
+            int: Number of tokens in the messages.
+        """
+        prompt = messages_to_prompt(messages, self.model_type)
```

Comment:
It seems `messages_to_prompt ` does not support anthropic models yet. How does it work?

https://github.com/camel-ai/camel/blob/916ab5eb5946ef3eb68a108b6dc0110375a76a9a/camel/utils/token_counting.py#L39